### ISSUE TITLE
Give DirectoryHelper.DeleteDirectory() a more meaningful name

### ISSUE THREAD

**Issue by nulltoken** [libgit2/libgit2sharp#980]
From https://github.com/libgit2/libgit2sharp/pull/977#issuecomment-75734210

>Second, would it make sense to rename the top-level DeleteDirectory() method into TryDeleteDirectory() as well? My rationale is that it (1) does not ensure directories are indeed deleted and (2) does not provide any direct feedback to the caller (only BaseFixture.Dispose() at the moment) that it could not delete a directory.

Why not? However, by convention, `TryXxxx` methods expose a certain signature (a returned `bool` and an in `out` param). I'd prefer using a name that doesn't conflict with a known pattern.

**Comment by ThomasBarnekow** [comment]
@nulltoken Having looked at the TryParse pattern, we could potentially do the following:
```
// This is a new method, wrapping the existing ones and making sure it does not throw.
// BaseFixture.Dispose() would call this one if we wanted to make sure tests don't fail.
// If you want failed tests as you said, Dispose would not call this one but the next one.
public static bool TryDeleteDirectory(string path);

// This is our current DeleteDirectory() method. It can theoretically throw. 
// Thus, we should keep the name.
public static void DeleteDirectory(string path);

// This is our current TryDeleteDirectory() method which throws under rare circumstances. 
// This is why I'd propose to change the name. If, and only if, you'd call TryDeleteDirectory() 
// from the Dispose() method, we could talk about moving that final trace message to 
// TryDeleteDirectory(). Otherwise, there would be no change.
private static void DeleteDirectory(string directoryPath, int maxAttempts, int initialTimeout, int timeoutFactor)
```

**Comment by nulltoken** [comment]
Fixed by #1022
